### PR TITLE
IdToFileMappingStrategy (issue #37)

### PR DESCRIPTION
**Pull request by ajurasz**
First proposal of IdToFileMappingStrategy. 

One thing to notice is that when the data is uploaded it is stored in the root of storage directory and after that it's moved to finall location provided by <code>IdToFileMappingStrategy#mapIdToFile</code>. Are we ok with this or should I work around this little bit more?

### CODE REVIEW

File: service/attachment/file/src/main/java/zed/service/attachment/file/strategy/CustomIdToFileMappingStrategy.java [hunk 1]
```diff
@@ -0,0 +1,15 @@
+package zed.service.attachment.file.strategy;
+
+import org.apache.commons.lang3.StringUtils;
+
+import java.io.File;
+import java.nio.file.Paths;
+
+public class CustomIdToFileMappingStrategy implements IdToFileMappingStrategy {
```
Comment (hekonsek) [inline service/attachment/file/src/main/java/zed/service/attachment/file/strategy/CustomIdToFileMappingStrategy.java]: `CustomIdToFileMappingStrategy` name doesn't tell much about the purpose of the class. I would rename it to `TenFirstLettersCustomIdToFileMappingStrategy` or something like that.

File: service/attachment/file/src/test/java/zed/service/attachment/file/strategy/CustomIdToFileMappingStrategyTest.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.SpringApplicationConfiguration;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+import zed.service.attachment.file.FileAttachmentServiceConfiguration;
+
+import java.io.File;
+
+@RunWith(SpringJUnit4ClassRunner.class)
```
Comment (hekonsek) [inline service/attachment/file/src/test/java/zed/service/attachment/file/strategy/CustomIdToFileMappingStrategyTest.java]: There's no need to start Spring context for this test. Remove:

     @RunWith(SpringJUnit4ClassRunner.class)
     @SpringApplicationConfiguration(classes = FileAttachmentServiceConfiguration.class)

And add replace...

    @Autowired
    IdToFileMappingStrategy idToFileMappingStrategy;

...with...

     IdToFileMappingStrategy idToFileMappingStrategy = new CustomIdToFileMappingStrategy();

That would make test faster (no need to bootstrap Spring Boot context).

**Comment by hekonsek** [comment]
Looks really good! Please just address two comments I left here a moment ago - then we can merge it. Good job :) .

**Comment by ajurasz** [comment]
Thank you for the review, I have addressed all your comments.

**Comment by hekonsek** [comment]
Yay! Merged. Many thanks for the good contribution! :)

**Pull request merged**